app/models/piece.rb

```diff
@@ -19,4 +19,19 @@ def move_to!(row, col)
     raise "Out of bounds" if row < 0 || row > 7 || col < 0 || col > 7
     update(row: row, col: col)
   end
+
+  def capture_piece(row, col)
+    other_piece = game.piece_at(row, col)
+    return true if game.path_obstructed? && other_piece.user != user
+    game.piece_at(row, col).captured!
```

Comment:
I think that an obstructed path should be part of piece movement and not knowledge required at the game controller level.

My thought is to have move_obstructed? as part of the valid_move? implementation. If we don't check for an obstructed path before updating the piece's location in the database record, the application is forced to do a database transaction rollback if either the path is obstructed or if the final location is obstructed (a piece of the same color is in the destination.)